packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js

```diff
@@ -315,33 +355,44 @@ export default class ReconnectionManager {
           this.meeting,
           {
             file: 'reconnection-manager/index',
-            function: 'executeReconnection'
+            function: 'rejoinMeeting'
           },
           EVENT_TRIGGERS.MEETING_STOPPED_SHARING_LOCAL,
           {
             reason: SHARE_STOPPED_REASON.MEETING_REJOIN
           }
         );
       }
-
-      return media;
     }
-    catch (error) {
-      // TODO: Add Metrics for Reconnect Failed
-      LoggerProxy.logger.error('ReconnectionManager:index#executeReconnection --> Media reestablishment failed');
-      this.status = RECONNECTION.STATE.FAILURE;
+    catch (joinError) {
+      this.rejoinAttempts += 1;
+      if (this.rejoinAttempts <= this.maxRejoinAttempts) {
+        LoggerProxy.logger.info(`ReconnectionManager:index#rejoinMeeting --> Unable to rejoin meeting, attempt #${this.rejoinAttempts}, retrying.`, joinError);
+        this.rejoinMeeting();
+      }
+      else {
+        LoggerProxy.logger.error('ReconnectionManager:index#rejoinMeeting --> Unable to rejoin meeting after max attempts.', joinError);
```

Comment:
I am not sure if we have to mark the rejoin failure as a reconnect failure , As its a totally different flow

Reply:
It is only a reconnect failure because it is part of the reconnection manager `reconnect` function. This doesn't send reconnect failure metrics, but to the end user, the reconnect has failed.